tests/virtual_machines_test.go

```diff
@@ -69,11 +69,42 @@ var _ = Describe("Virtual Machines", func() {
 				return len(vmList.Items)
 
 			}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
+
+			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
+				err = cleanNamespaceLabels(namespace)
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		Context("When the client wants to create a vm on an opted-out namespace", func() {
+			It("should create a vm object without kubemacpool MAC address assignment when no label is set to namespace", func() {
+				err := initKubemacpoolParams(rangeStart, rangeEnd, map[string]string{})
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
+					[]kubevirtv1.Network{newNetwork("br")})
+
+				err = testClient.VirtClient.Create(context.TODO(), vm)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).Should(Equal(""))
+			})
+
+			It("should create a vm object without kubemacpool MAC address assignment when disabled label is set to namespace", func() {
```

Comment:
This is a different context. Context describes what is the environment Context: "When KMP is disabled on a namespace", It: "should create a VM object without a MAC assigned".